mesonbuild/minstall.py

```diff
@@ -321,6 +321,7 @@ def do_copydir(self, data, src_dir, dst_dir, exclude, install_mode):
                 abs_dst = os.path.join(dst_dir, filepart)
                 if os.path.isdir(abs_dst):
                     print('Tried to copy file %s but a directory of that name already exists.' % abs_dst)
+                    sys.exit(1)
```

Comment:
I think it should raise MesonException() instead? There seems to be a mix of `raise RuntimeError` and `exit(1)` in minstall.py. But probably could go as follow-up cleanup.

Reply:
minstall.py does not import `MesonException`. That sort of rewrite should probably happen separately. 

The difference between `RuntimeError` and `print()` + `exit(1)` is that the former is for programming errors and the latter is for expected errors.

Reply:
That probably go together with using mlog, because raising MesonException is actually going to use mlog in mesonmain.py.